compiler/src/yul/runtime/functions/data.rs

```diff
@@ -380,3 +381,29 @@ pub fn load_data_string() -> yul::Statement {
         }
     }
 }
+
+/// Revert with encoded reason string
+pub fn revert_with_reason_string() -> yul::Statement {
+    function_definition! {
+        function revert_with_reason_string(reason) {
+            (let ptr := alloc(4))
+            // Function selector for Error(string)
+            (mstore(ptr, 3963877391197344453575983046348115674221700746820753546331534351508065746944))
+
+            // Write the (fixed) data offset into the next 32 bytes of memory
+            (let __ := alloc_mstoren(0x0000000000000000000000000000000000000000000000000000000000000020, 32))
+
+            // Read the size of the string
+            (let reason_size := mloadn(reason, 32))
+
+            //Copy the whole reason string (length + data) to the current segment of memory
+            (__ := mcopym(reason , (add(reason_size, 32))))
```

Comment:
We can also `pop` these values right off the stack.

`(pop((mcopym(reason , (add(reason_size, 32))))))`

All function arguments are matched as token trees, so expressions like `mcopym(reason , (add(reason_size, 32)))` need to be wrapped in parens. Biggest downside with these macros imo.

Reply:
Ah, right, thanks. Does it make any practical difference whether to assign them to an unused variable or pop them?

Reply:
The optimizer will probably convert these to pops anyway.

Even if we weren't running the optimizer, I don't think it would make any difference in terms of gas. We would just keep these values on the stack a bit longer.